source/features/default-to-rich-diff.tsx

```diff
@@ -0,0 +1,47 @@
+import select from 'select-dom';
+import features from '../libs/features';
+
+const childList = {childList: true};
+
+// To push new extensions if we want more file types to have rich diff as default
+const richDiffExtensions = ['svg'];
+
+const fileExtensionRegExp = new RegExp('\\.(' + richDiffExtensions.join('|') + ')$');
+
+const checkFileIsRichDiffable = (filename: string): boolean => fileExtensionRegExp.test(filename);
+
+const observer = new MutationObserver(([{addedNodes}]) => {
+	for (const node of addedNodes) {
+		const element = node as Element;
+		if (element.tagName === 'DIV') {
+			if (element.className === 'js-diff-progressive-container') {
```

Comment:
You can merge these two conditions into `element.matches('.js-diff-progressive-container')`

Reply:
Some of the added nodes can be text nodes and, in that case, `matches` is not supported, I believe.

Reply:
You can skip the `as` and use `node instanceof Element && element.matches(…)`